src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpCollectionExpressionRewriter.cs

```diff
@@ -707,4 +741,34 @@ static StatementSyntax UnwrapEmbeddedStatement(StatementSyntax statement)
             return arguments.SelectAsArray(a => indent(a.Expression));
         }
     }
+
+    /// <summary>
+    /// Use the same EOL text when producing the collection as the EOL on the line the original expression was on.
+    /// </summary>
+    private static SyntaxTrivia DetermineEndOfLine<TParentExpression>(
+        ParsedDocument document, TParentExpression expressionToReplace, SyntaxFormattingOptions formattingOptions) where TParentExpression : ExpressionSyntax
+    {
+        var text = document.Text;
+        var lineToConsider = text.Lines.GetLineFromPosition(expressionToReplace.SpanStart);
+        var lineBreakSpan = TextSpan.FromBounds(lineToConsider.End, lineToConsider.EndIncludingLineBreak);
+
+        return lineBreakSpan.IsEmpty
+            ? EndOfLine(formattingOptions.NewLine)
+            : EndOfLine(text.ToString(lineBreakSpan));
+    }
+
+    private static SyntaxToken RemoveTrailingWhitespace(SyntaxToken token)
```

Comment:
💭 I would have expected this to already exist, but maybe it's just in StyleCop Analyzers

Reply:
yeah, i looked in many places and didn't find it.  i even looked for all our usages of SyntaxKind.WhitespaceTrivia.  We have other manipulation functions elsewhere, but nothing that just trims whitespace.